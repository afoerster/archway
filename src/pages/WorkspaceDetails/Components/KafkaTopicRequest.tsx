import * as React from 'react';
import { Form } from 'antd';
import { Field, InjectedFormProps } from 'redux-form';
import { reduxForm } from 'redux-form/immutable';

/* tslint:disable:no-var-requires */
const { TextField } = require('redux-form-antd');

interface KafkaForm {
  name: string;
}

const KafkaTopicRequest = ({ handleSubmit }: InjectedFormProps<KafkaForm, {}>) => (
  <Form style={{  }} onSubmit={handleSubmit}>
    <Field name="name" component={TextField} />
  </Form>
);

export default reduxForm<KafkaForm, {}>({
  form: 'topicRequest',
})(KafkaTopicRequest);
