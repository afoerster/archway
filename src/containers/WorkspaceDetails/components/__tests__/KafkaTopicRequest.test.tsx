import * as React from 'react';
import { shallow } from 'enzyme';

import KafkaTopicRequest from '../KafkaTopicRequest';

describe('KafkaTopicRequest', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<KafkaTopicRequest />);
    expect(wrapper).toMatchSnapshot();
  });
});
