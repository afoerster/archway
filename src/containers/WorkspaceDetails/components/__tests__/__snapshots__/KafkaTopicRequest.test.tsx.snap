// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KafkaTopicRequest renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    destroyOnUnmount={true}
    enableReinitialize={false}
    forceUnregisterOnUnmount={false}
    form="topicRequest"
    getFormState={[Function]}
    keepDirtyOnReinitialize={false}
    persistentSubmitErrors={false}
    pure={true}
    shouldAsyncValidate={[Function]}
    shouldError={[Function]}
    shouldValidate={[Function]}
    shouldWarn={[Function]}
    touchOnBlur={true}
    touchOnChange={false}
    updateUnregisteredFields={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "destroyOnUnmount": true,
      "enableReinitialize": false,
      "forceUnregisterOnUnmount": false,
      "form": "topicRequest",
      "forwardedRef": null,
      "getFormState": [Function],
      "keepDirtyOnReinitialize": false,
      "persistentSubmitErrors": false,
      "pure": true,
      "shouldAsyncValidate": [Function],
      "shouldError": [Function],
      "shouldValidate": [Function],
      "shouldWarn": [Function],
      "touchOnBlur": true,
      "touchOnChange": false,
      "updateUnregisteredFields": false,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "destroyOnUnmount": true,
        "enableReinitialize": false,
        "forceUnregisterOnUnmount": false,
        "form": "topicRequest",
        "forwardedRef": null,
        "getFormState": [Function],
        "keepDirtyOnReinitialize": false,
        "persistentSubmitErrors": false,
        "pure": true,
        "shouldAsyncValidate": [Function],
        "shouldError": [Function],
        "shouldValidate": [Function],
        "shouldWarn": [Function],
        "touchOnBlur": true,
        "touchOnChange": false,
        "updateUnregisteredFields": false,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
